BookAPI/Controllers/BooksController.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace BookAPI.Controllers
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using BookAPI.Services;
+    using BookAPI.Models;
+    using Microsoft.AspNetCore.Mvc;
+
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookShelf _books;
+
+        public BooksController(IBookShelf books)
+        {
+            _books = books;
+        }
+
+        [HttpPost]
+        public ActionResult<Book> PostBook(Book book)
```

Comment:
IActionResult will be easier to use, check the link in the chat. This is also ok.